### DISCUSSION TITLE
Is it possible to use  openzeppelin-upgradable contracts ?

### DISCUSSION THREAD

**Discussion by mariolim96** [scaffold-eth/scaffold-eth-2#673] [Q&A] [answered]
I'm working on a project with various smart contract and different proxies(like uups, beacon and so on).
I really like the ui debug system and i would like to use it to test the interactions between the smart contract. 
I've seen that hardhat-deploy and openzeppelin doesn't work together.
There is a way to use openzeppelin upgrades contract into this template ?

I was thinking to remove hardhat-deploy and 
and create some script that generates me the deployment folder.

The ui can interact with uups contracts?

**Comment by technophile-04** [comment, marked as answer]
Actually this is not baked in into SE-2 out of the box, but you could do it easily...checkout this demo video where I tried linking `hardhat-deploy` openzepllin upgrades and SE-2 : https://youtu.be/cdNyMOrw3uc?si=MTS7xmnWewiNbIG0&t=1267  (watch it on 1.5x)

Here is the repo : https://github.com/technophile-04/uups-proxy/tree/main/packages/hardhat

Mainly checkout deploy script : https://github.com/technophile-04/uups-proxy/blob/main/packages/hardhat/deploy/00_deploy_yourContract.ts

and upgrade script : https://github.com/technophile-04/uups-proxy/blob/main/packages/hardhat/scripts/upgrateToV2.ts

I think the demo video should give you the idea how it works but please let me know if you need any help 🙌

**Reply by mariolim96** [reply]
I want to use openzepplin v5, do I have to make any particular changes?

**Reply by technophile-04** [reply]
I think it should be fine, since not a lot of breaking changes in oz-upgrades
